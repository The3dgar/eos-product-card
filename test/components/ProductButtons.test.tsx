import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src';
import { product1 } from '../data/product';

describe('ProductTitle', () => {
  test('should show custom title', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Custom title" className="custom-class" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show the component with product name (context)', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          ()=><ProductTitle />  
        }
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
