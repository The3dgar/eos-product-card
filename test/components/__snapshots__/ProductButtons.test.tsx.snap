// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should show custom title 1`] = `
<span
  className="productDescription custom-class"
>
  Custom title
</span>
`;

exports[`ProductTitle should show the component with product name (context) 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee mug - card
  </span>
</div>
`;
